build-tasks/create-rule.js

```diff
@@ -1,79 +1,145 @@
-    const USAGE_EXAMPLE = '\nUsage example:\nnpm run create-rule -- --rule-name=no-something-or-other\n';
-
-    if (!ruleName) {
-        console.log(red('--rule-name parameter is required.' + USAGE_EXAMPLE));
-        process.exit(1);
+const inquirer = require('inquirer');
+const { execSync } = require('child_process');
+const { writeFile } = require('./common/files');
+
+const questions = [
+    {
+        name: 'name',
+        message: 'Name:',
+        type: 'input',
+        validate: value => {
+            if (!/^[a-z0-9]+(\-[a-z0-9]+)*$/.test(value)) {
+                return 'The name should consist of lowercase letters and numbers separated with "-" character.';
+            }
+
+            return true;
+        }
+    },
+    {
+        name: 'description',
+        message: 'Description:',
+        type: 'input',
+        validate: value => {
+            if (!!value && !!value.trim()) {
+                return true;
+            }
+            return 'Please enter a description for the rule.';
+        }
+    },
+    {
+        name: 'type',
+        message: 'Rule type:',
+        type: 'list',
+        choices: ['functionality', 'maintainability', 'style', 'typescript'],
+        default: 'maintainability'
+    },
+    {
+        name: 'issueClass',
+        message: 'Issue class:',
+        type: 'list',
+        choices: ['SDL', 'Non-SDL', 'Ignored'],
+        default: 'Non-SDL'
+    },
+    {
+        name: 'issueType',
+        message: 'Issue type:',
+        type: 'list',
+        choices: ['Error', 'Warning'],
+        default: 'Warning'
+    },
+    {
+        name: 'severity',
+        message: 'Severity:',
+        type: 'list',
+        choices: ['Critical', 'Important', 'Moderate', 'Low'],
+        default: 'Low'
+    },
+    {
+        name: 'level',
+        message: 'Level:',
+        type: 'list',
+        choices: ['Mandatory', 'Opportunity for Excellence'],
+        default: 'Opportunity for Excellence'
+    },
+    {
+        name: 'group',
+        message: 'Group:',
+        type: 'list',
+        choices: ['Clarity', 'Configurable', 'Correctness', 'Deprecated', 'Ignored', 'Security', 'Whitespace'],
+        default: 'Clarity'
     }
-
-    if (!/^[a-z0-9]+(\-[a-z0-9]+)*$/.test(ruleName)) {
-        console.log(red('Rule name should consist of lowercase letters and numbers separated with "-" character.' + USAGE_EXAMPLE));
-        process.exit(1);
+];
+
+inquirer.prompt(questions).then(answers => {
+    const sourceFileName = createImplementationFile(answers);
+    const testFileName = createTestFiles(answers.name);
+
+    console.log(`Rule '${answers.name}' created.`);
+    console.log(`Source file: ${sourceFileName}`);
+    console.log(`Test file:   ${testFileName}`);
+
+    // If we're running in the VS Code terminal, try to open the
+    // new files. If we can't do it, then it's not a big deal.
+    if (process.env.VSCODE_CWD) {
+        try {
+            execSync(`code "${testFileName}"`);
```

Comment:
Is there an easy way to tell if it is `code` or `code-insiders`? In my case it will open separate instance of VS Code instead of reusing current VS Code insiders window.

May be use `inquirer` to prompt if user want to open files in case when `VSCODE_CWD` present (conditionally add question to initial list)?

Reply:
Oh, nice catch. I hadn't even thought about that. Looks like there's a `TERM_PROGRAM_VERSION` environment variable we can use to check if the version of VS Code is the Insiders version.